File: daily/双指针/双指针hard/42. 接雨水.js
/**
 * @param {number[]} height
 * @return {number}
 */
var trap = function(height) {
    let sum = 0
    // 第一列和最后一列不会积水
    for(let i=1; i<height.length-1; i++) {
        let max_left = 0    // 找到左边最高柱子
        for(let j=0; j<i; j++)
            if(height[j] > max_left)
                max_left = height[j]
        let max_right = 0   // 找到右边最高柱子
        for(let j=i+1; j<height.length; j++)
            if(height[j] > max_right)
                max_right = height[j]
        if(Math.min(max_left, max_right) > height[i])   // 短板效应
            sum += Math.min(max_left, max_right) - height[i]
    }
    return sum
};

File: daily/C++版本/42. 接雨水.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        int size=height.size(),sum=0;
        // 第一列和最后一列不会积水
        for(int i=1;i<size-1;i++){
        	int max_left=0;
            for(int j=0;j<i;j++)   //求左边最高的柱子 
                if(max_left<height[j])
                    max_left=height[j];
            int max_right=0;
            for(int j=i+1;j<size;j++)  //右边最高的柱子
                if(max_right<height[j])
                    max_right=height[j];
            if(min(max_left,max_right)>height[i])	//短板效应，然后比当前列高才能积水 
            	sum+=min(max_left,max_right)-height[i];
        } 
        return sum;
    }
};
